File: test-programs/elixir/canonical_flow/lib/canonical_flow.ex
defmodule CanonicalFlow do
  @moduledoc false

  def compute do
    a = 10
    b = 32
    sum_val = a + b
    doubled = sum_val * 2
    final_result = doubled + a
    true = final_result == 94
    final_result
  end

  def main do
    result = compute()
    true = result == 94
    IO.puts(result)
    result
  end
end

File: test-programs/erlang/canonical_flow/src/canonical_flow.erl
-module(canonical_flow).
-export([compute/0, main/0]).

compute() ->
    A = 10,
    B = 32,
    SumVal = A + B,
    Doubled = SumVal * 2,
    FinalResult = Doubled + A,
    true = FinalResult =:= 94,
    FinalResult.

main() ->
    Result = compute(),
    true = Result =:= 94,
    io:format("~p~n", [Result]),
    Result.
